webapp/tests/karma/ts/effects/contacts.effects.spec.ts

```diff
@@ -245,6 +245,53 @@ describe('Contacts effects', () => {
         ]]);
       });
 
+      it('should not receive children if the selected contact changes', async () => {
+        contactViewModelGeneratorService.getContact.onFirstCall()
+          .resolves({_id: 'contact1', doc: {_id: 'contact1'}});
+        contactViewModelGeneratorService.getContact.onSecondCall()
+          .resolves({_id: 'contact2', doc: {_id: 'contact2'}});
+        contactViewModelGeneratorService.loadChildren.onFirstCall().callsFake(async () => {
+          // Change the selected contact before returning the first set of children.
+          actions$ = of(ContactActionList.selectContact({id: 'contact2'}));
+          await effects.selectContact.toPromise();
+          return Promise.resolve([{type: {id: 'person'}, contacts: [{_id: 'person1'}]}]);
```

Comment:
I _think_ that, since you have an await in this function, the returning value is already a promise. so you just need to return the value, without the promise wrapper. Makes the code a bit more readable.  
```suggestion
          return [{ type: { id: 'person' }, contacts: [{ _id: 'person1' }] }];
```